sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -44,10 +44,10 @@ abstract class RDG extends UnaryExpression with ExpectsInputTypes with Stateful
   }
 
   @transient protected lazy val seed: Long = child match {
-    case Literal(s, IntegerType) => s.asInstanceOf[Int]
-    case Literal(s, LongType) => s.asInstanceOf[Long]
+    case e if child.foldable && e.dataType == IntegerType => e.eval().asInstanceOf[Int]
+    case e if child.foldable && e.dataType == LongType => e.eval().asInstanceOf[Long]
     case _ => throw new AnalysisException(
-      s"Input argument to $prettyName must be an integer, long or null literal.")
+      s"Input argument to $prettyName must be an integer or long constant.")
```

Comment:
Like Apache Spark 3.0.1, it seems that we support still `null` here. If we didn't exclude it, shall we keep to mention `null`?
```
scala> sql("select rand(null)").show()
+-----------------------+
|rand(CAST(NULL AS INT))|
+-----------------------+
|     0.7604953758285915|
+-----------------------+
```

Reply:
okay